project/ScalacOptions.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import sbt._, Keys._
+
+object Scalac {
+
+  // see: https://tpolecat.github.io/2017/04/25/scalac-flags.html
+  val extraOptions = List(
+    "-encoding", "utf-8", // Specify character encoding used by source files.
+    "-explaintypes", // Explain type errors in more detail.
+    // "-language:existentials", // Existential types (besides wildcard types) can be written and inferred
+    // "-language:experimental.macros", // Allow macro definition (besides implementation and application)
+    // "-language:higherKinds", // Allow higher-kinded types
```

Comment:
can we also enable `higherKinds` since we use it in a few places?

Reply:
We could. Right now we import the features when we need them, but that's something we can change.